docs/Golden-Gate/Triggers.rst

```diff
@@ -0,0 +1,108 @@
+Simulation Triggers
+===================
+
+It is often useful to globally coordinate debug and instrumentation features using specific target-events that may be distributed across the target design.
+For instance, you may wish to enable collection of synthesized prints and sampling of
+AutoCounters simulataenously when a specific instruction is committed on any
+core, or alternatively if the memory system sees a ready or write to a
+particular memory address. Golden Gate's trigger system enables this by aggregating annotated ``TriggerSources``
+distributed throughout the design using a centralized credit-based system
+which then drives a single-bit level-sensitive enable to all ``TriggerSinks`` sinks distributed throughout the design.
+This enable is asserted while the design remains in the region-of-interested
+(ROI).  Sources signal the start of the ROI by granting a
+credit and signal the end of the ROI by asserting a debit. Since there can be
+multiple sources, each of which might grant credits, the trigger is only
+disabled when the system has been debited as exactly as many times as it has
+been credited (it has a balance of 0).
+
+Chisel API
+-----------
+Trigger sources and sinks are Boolean signals, synchronous to a particular
+clock domain, that have been annotated as such.  The ``midas.targetutils``
+package provides chisel-facing utilities for annotating these signals in your
+design. We describe these utilities below, the source for which can be found in
+``sim/midas/targetutils/src/main/scala/annotations.scala``.
+
+Trigger Sources
+***************
+In order to permit distributing trigger sources across the whole design, you
+must annotate distinct boolean signals as credits and debits using methods
+provided by the ``TriggerSource`` object. We provide an example below.
+
+.. literalinclude:: ../../sim/src/main/scala/midasexamples/TriggerWiringModule.scala
+    :language: scala
+    :start-after: DOC include start: TriggerSource Usage
+    :end-before: DOC include end: TriggerSource Usage
+
```

Comment:
This example is fine for very advanced users, but I think it might also be worth also having a simpler example that doesn't use the credit and debit terminology.
To be frank, if I wanted to re-implement the TracerV trigger based on reading these docs and this example, I don't think I would be able to understand how to do it.
What would help me is having an example that just uses a single trigger boolean signal (with no credit and debit terminology), that I essentially want to "magically and deterministically" (this is where the GG magic happens using this entire credit/debit system) wire to the Sink.